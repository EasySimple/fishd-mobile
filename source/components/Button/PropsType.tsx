export interface ButtonPropsType {
  type?: 'primary' | 'guide' | 'default' | 'warning' | 'danger';
  size?: 'large' | 'normal' | 'small' | 'mini';
  plain?: boolean;
  color?: string;
  round?: boolean;
  square?: boolean;
  disabled?: boolean;
  loading?: boolean;
}
