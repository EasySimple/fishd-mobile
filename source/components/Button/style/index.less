@import (reference) '../../../assets/css/index';

@buttonPrefixCls: fm-button;
@guideColor: #26BD71;
@dangerColor: #EE0A24;
@warningColor: #FFAF0F;
@primaryColor: #337EFF;

.plain(@plainColor) {
  background-color: #fff;
  color: @plainColor;
  border: 1px solid @plainColor;
}

.@{buttonPrefixCls} {
  outline: 0 none;
  -webkit-appearance: none;
  box-sizing: border-box;
  border-radius: 2px;
  text-align: center;
  font-size: 14px;
  letter-spacing: 0;
  height: 44px;
  line-height: 1.2;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  text-overflow: ellipsis;
  word-break: break-word;
  white-space: nowrap;

  &&-active {
    position: relative;
    &::before {
      content: '';
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-color: #000;
      opacity: 0.1;
    }
  }

  &&-hairline {
    border-width: 0!important;
    position: relative;
    &::after {
      display: block;
      content: '';
      position: absolute;
      top: -50%;
      right: -50%;
      bottom: -50%;
      left: -50%;
      border: 1px solid #ebedf0;
      border-color: inherit;
      transform: scale(0.5);
    }
  }

  &&-block {
    display: flex;
    width: 100%;
  }

  &&-disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }

  &-default {
    color: #323233;
    background-color: #fff;
    border: 1px solid #ebedf0;
  }

  &-primary {
    color: #fff;
    background-color: @primaryColor;
    &.@{buttonPrefixCls}-plain {
      .plain(@primaryColor);
    }
  }

  &-guide {
    color: #fff;
    background-color: @guideColor;
    &.@{buttonPrefixCls}-plain {
      .plain(@guideColor);
    }
  }

  &-danger {
    color: #fff;
    background-color: @dangerColor;
    &.@{buttonPrefixCls}-plain {
      .plain(@dangerColor);
    }
  }

  &-warning {
    color: #fff;
    background-color: @warningColor;
    &.@{buttonPrefixCls}-plain {
      .plain(@warningColor);
    }
  }

  &-round {
    border-radius: 999px;
  }

  &-square {
    border-radius: 0;
  }

  &-large {
    width: 100%;
    height: 50px;
  }

  &-normal {
    padding: 0 20px;
    font-size: 14px;
  }

  &-small {
    height: 32px;
    padding: 0 12px;
    font-size: 12px;
  }

  &-mini {
    height: 24px;
    padding: 0 8px;
    font-size: 10px;
  }

  .fm-icon + .fm-button-text {
    margin-left: 4px;
  }

  > .@{buttonPrefixCls}-icon {
    margin-right: 0.5em;
  }
}
