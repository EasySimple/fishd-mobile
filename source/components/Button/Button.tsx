import classnames from 'classnames';
import * as React from 'react';
import TouchFeedback from 'rmc-feedback';
import { ButtonPropsType } from './PropsType';
import './style/index.less';

export interface ButtonProps extends ButtonPropsType {
  prefixCls?: string;
  className?: string;
  role?: string;
  inline?: boolean;
  icon?: React.ReactNode;
  activeClassName?: string;
  activeStyle?: boolean | React.CSSProperties;
  style?: React.CSSProperties;
  onClick?: React.MouseEventHandler<HTMLAnchorElement>;
}

class Button extends React.Component<ButtonProps, any> {
  static defaultProps = {
    prefixCls: 'fm-button',
    size: 'large',
    inline: false,
    disabled: false,
    loading: false,
    activeStyle: {},
  };

  render() {
    const {
      children,
      className,
      prefixCls,
      type,
      size,
      inline,
      disabled,
      loading,
      activeStyle,
      activeClassName,
      onClick,
      ...restProps
    } = this.props;

    const wrapCls = classnames(prefixCls, className, {
      [`${prefixCls}-primary`]: type === 'primary',
      [`${prefixCls}-ghost`]: type === 'ghost',
      [`${prefixCls}-warning`]: type === 'warning',
      [`${prefixCls}-small`]: size === 'small',
      [`${prefixCls}-inline`]: inline,
      [`${prefixCls}-disabled`]: disabled,
      [`${prefixCls}-loading`]: loading,
    });

    return (
      <TouchFeedback
        activeClassName={activeClassName || (activeStyle ? `${prefixCls}-active` : undefined)}
        disabled={disabled}
        activeStyle={activeStyle}
      >
        <a
          role="button"
          className={wrapCls}
          {...restProps}
          onClick={disabled ? undefined : onClick}
          aria-disabled={disabled}
        >
          {children}
        </a>
      </TouchFeedback>
    );
  }
}

export default Button;
