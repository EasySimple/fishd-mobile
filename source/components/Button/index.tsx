import Button from './Button';
import ButtonGroup from './ButtonGroup';
import './style/index.less';

export { ButtonProps, ButtonShape, ButtonSize, ButtonType } from './Button';
export { ButtonGroupProps } from './ButtonGroup';

Button.Group = ButtonGroup;
export default Button;
