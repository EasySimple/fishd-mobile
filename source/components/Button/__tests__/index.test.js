import React from 'react'
import { shallow, render, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import Button from '../Button'

describe('<Button />', () => {
test('<Button /> render text', () => {
    const wrapper = mount(<Button>button text</Button>)
    expect(wrapper.find('.fm-button').text()).toBe('button text')
  })

test('should call onClick callback if provided', () => {
    const onClickMock = jest.fn();
    const wrapper = mount(<Button onClick={onClickMock}>hello</Button>);
    wrapper.find('.fm-button').simulate('click');
    expect(onClickMock).toHaveBeenCalled();
  });

test('<Button /> snapshot', () => {
    const largeWrapper = render(<Button size='large' />)
    expect(toJson(largeWrapper)).toMatchSnapshot()

    const smallWrapper = render(<Button size='large'>hello world</Button>)
    expect(toJson(smallWrapper)).toMatchSnapshot()
  })
})
