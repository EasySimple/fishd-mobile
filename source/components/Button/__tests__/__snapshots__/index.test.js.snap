// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> <Button /> snapshot 1`] = `
<a
  aria-disabled="false"
  class="fm-button"
  role="button"
/>
`;

exports[`<Button /> <Button /> snapshot 2`] = `
<a
  aria-disabled="false"
  class="fm-button"
  role="button"
>
  hello world
</a>
`;
